src/Microsoft.HttpRepl/Program.cs

```diff
@@ -18,7 +20,9 @@ class Program
         static async Task Main(string[] args)
         {
             IFileSystem fileSystem = new RealFileSystem();
-            HttpState state = new HttpState(fileSystem);
+            IUserProfileDirectoryProvider userProfileDirectoryProvider = new UserProfileDirectoryProvider();
+            IPreferencesProvider preferencesProvider = new PreferencesProvider(fileSystem, userProfileDirectoryProvider);
+            HttpState state = new HttpState(fileSystem, preferencesProvider);
```

Comment:
nit: Can we create GetHttpState(..) and move everything above there?

Reply:
I moved everything except the IFileSystem into a function, since that's used elsewhere in this function.